lib/spack/spack/binary_distribution.py

```diff
@@ -37,23 +39,294 @@
 from spack.spec import Spec
 from spack.stage import Stage
 
+
+#: default root, relative to the Spack install path
+default_manager_root = os.path.join(spack.paths.opt_path, 'spack')
+
 _build_cache_relative_path = 'build_cache'
 _build_cache_keys_relative_path = '_pgp'
 
-BUILD_CACHE_INDEX_TEMPLATE = '''
-<html>
-<head>
-  <title>{title}</title>
-</head>
-<body>
-<ul>
-{path_list}
-</ul>
-</body>
-</html>
-'''
 
-BUILD_CACHE_INDEX_ENTRY_TEMPLATE = '  <li><a href="{path}">{path}</a></li>'
+class BinaryDistributionCacheManager(object):
+    """
+    The BinaryDistributionCacheManager stores and manages both:
+
+    1. cached buildcache index files (the index.json describing the built
+    specs available on a mirror)
+
+    2. a cache of all the conrcrete built specs available on all the
+    configured mirrors.
+    """
+
+    def __init__(self, cache_root=None):
+        self._cache_root = cache_root or default_manager_root
+        self._index_cache_root = os.path.join(self._cache_root, 'indices')
+        self._index_contents_key = 'contents.json'
+        self._index_file_cache = None
+        self._local_index_cache = None
+        self._built_spec_cache = {}
```

Comment:
For a given DAG hash, this contains a list of all binary caches which have a binary package for that DAG hash. I think this member could use a comment explaining that or an updated name like `self._mirrors_for_spec`.